app/actions/remote/channel.ts

```diff
@@ -394,38 +394,33 @@ export const fetchMissingSidebarInfo = async (serverUrl: string, directChannels:
         }
     });
 
-    const filteredUserIds = new Set(users.map((u) => u.id));
     if (currentUserId) {
         const ownDirectChannel = directChannels.find((dm) => dm.name === getDirectChannelName(currentUserId, currentUserId));
         const database = DatabaseManager.serverDatabases[serverUrl]?.database;
         if (ownDirectChannel && database) {
             const currentUser = await getCurrentUser(database);
             ownDirectChannel.display_name = displayUsername(currentUser, locale, teammateDisplayNameSetting, false);
         }
-        filteredUserIds.add(currentUserId);
     }
 
-    const profiles = users.reduce((acc: UserProfile[], u) => {
-        if (!filteredUserIds.has(u.id)) {
-            acc.push(u);
-        }
-        return acc;
-    }, []);
-
     if (!fetchOnly) {
         const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
         if (operator) {
             const modelPromises: Array<Promise<Model[]>> = [];
             if (users.length) {
-                modelPromises.push(operator.handleUsers({users: profiles, prepareRecordsOnly: true}));
+                modelPromises.push(operator.handleUsers({users, prepareRecordsOnly: true}));
+            }
+
+            if (shouldUpdate) {
```

Comment:
Maybe I am missing something, but this seems to be true the moment we push an element to users.length. Therefore `Boolean(users.length)===shouldUpdate`, so we can remove the shouldUpdate, right?